core/include/webview/detail/engine_base.hh

```diff
@@ -23,132 +23,97 @@
  * SOFTWARE.
  */
 
-#ifndef WEBVIEW_DETAIL_ENGINE_BASE_HH
-#define WEBVIEW_DETAIL_ENGINE_BASE_HH
-
+#pragma once
 #if defined(__cplusplus) && !defined(WEBVIEW_HEADER)
 
-#include "../errors.hh"
 #include "../types.h"
 #include "../types.hh"
-#include "json.hh"
 #include "user_script.hh"
-
 #include <atomic>
-#include <functional>
 #include <list>
 #include <map>
-#include <string>
+
+using binding_t = std::function<void(std::string, std::string, void *)>;
+using sync_binding_t = std::function<std::string(std::string)>;
 
 namespace webview {
 namespace detail {
 
+using noresult = webview::noresult;
```

Comment:
This is a brevity convenience to put `noresult` into the `webview::detail` namespace.

llvm does not like any ambiguity in namespacing, so using both `webview` and `webview::detail` in `engine_base.cc` is a compiler fail. I was using gcc locally, which has no issues, and you can see a few commits in this PR while I was trying to figure out what was going on...